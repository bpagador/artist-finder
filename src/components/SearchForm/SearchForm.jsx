import React from 'react';
import PropTypes from 'prop-types';

function SearchForm({ artistName, onChange, onSubmit }) {
  return (
    <form onSubmit={onSubmit}>
      <label>
        Search for Artist
        <input type="text" name="artistName" value={artistName} onChange={onChange}></input>
      </label>
      <button>Submit</button>
    </form>
  );
}

SearchForm.propTypes = {
  artistName: PropTypes.string.isRequired,
  onChange: PropTypes.func.isRequired,
  onSubmit: PropTypes.func.inRequired
};

export default SearchForm;


