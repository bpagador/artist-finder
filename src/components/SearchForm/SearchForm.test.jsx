import React from 'react';
import { shallow } from 'enzyme';
import SearchForm from './SearchForm';

describe('SearchForm component', () => {
  let handleChange;
  let handleSubmit;
  let wrapper;

  beforeEach(() => {
    handleChange = jest.fn();
    handleSubmit = jest.fn();
    wrapper = shallow(<SearchForm
      artistSearch="beyonce"
      onChange={handleChange}
      onSubmit={handleSubmit}
    />);
  });

  it('has an input with a value set to artistSearch', () => {
    expect(wrapper.find('input[name="artistSearch"]').prop('value')).toEqual('beyonce');
  });

  it('invokes an onChange prop when the artistSearch input changes', () => {
    wrapper.find('input[name="artistSearch"]').simulate('change');
    expect(handleChange).toHaveBeenCalledTimes(1);
  });

  it('invokes an onSubmit prop when submit button is clicked', () => {
    wrapper.find('form').simulate('submit');
    expect(handleSubmit).toHaveBeenCalledTimes(1);
  });

  it('renders the SearchForm component', () => {
    expect(wrapper).toMatchSnapshot();
  });
});
