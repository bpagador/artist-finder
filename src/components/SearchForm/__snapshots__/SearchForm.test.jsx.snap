// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchForm component renders the SearchForm component 1`] = `
<form
  onSubmit={[MockFunction]}
>
  <label>
    Search for Artist
    <input
      name="artistName"
      onChange={[MockFunction]}
      type="text"
      value="beyonce"
    />
  </label>
  <button>
    Submit
  </button>
</form>
`;
